xcodeproj/internal/target.bzl

```diff
@@ -46,350 +39,24 @@ load(
-        product_name = props.product_name,
-        product_type = props.product_type,
-        bundle_path = props.bundle_path,
-        linker_inputs = linker_inputs,
-        build_settings = build_settings,
-    )
-
-    resource_bundles = resource_bundle_products.collect(
-        owner = resource_owner,
-        is_consuming_bundle = is_bundle,
-        bundle_resources = bundle_resources,
-        attrs_info = attrs_info,
-        transitive_infos = transitive_infos,
-    )
-
-    cc_info = target[CcInfo] if CcInfo in target else None
-    objc = target[apple_common.Objc] if apple_common.Objc in target else None
-    _process_defines(
-        cc_info = cc_info,
-        build_settings = build_settings,
-    )
-    _process_sdk_links(
-        objc = objc,
-        build_settings = build_settings,
-    )
-    search_paths = _process_search_paths(
-        cc_info = cc_info,
-        objc = objc,
-        opts_search_paths = opts_search_paths,
-    )
-
-    return processed_target(
-        attrs_info = attrs_info,
-        dependencies = dependencies,
-        inputs = inputs,
-        linker_inputs = linker_inputs,
-        outputs = outputs,
-        potential_target_merges = potential_target_merges,
-        required_links = required_links,
-        resource_bundles = resource_bundles,
-        search_paths = search_paths,
-        target = struct(
-            id = id,
-            label = label,
-            is_bundle = is_bundle,
-            product_path = product.path,
-        ),
-        xcode_target = xcode_target(
-            id = id,
-            name = ctx.rule.attr.name,
-            label = label,
-            configuration = configuration,
-            package_bin_dir = package_bin_dir,
-            platform = platform,
-            product = product,
-            is_bundle = is_bundle,
-            is_swift = is_swift,
-            test_host = (
-                test_host_target_info.target.id if test_host_target_info else None
-            ),
-            avoid_infos = [
-                ("test_host", dep[XcodeProjInfo])
-                for dep in avoid_deps
-            ],
-            build_settings = build_settings,
-            search_paths = search_paths,
-            modulemaps = modulemaps,
-            swiftmodules = _process_swiftmodules(swift_info = swift_info),
-            resource_bundles = resource_bundles,
-            inputs = inputs,
-            linker_inputs = linker_inputs,
-            info_plist = info_plist,
-            entitlements = entitlements_file_path,
-            dependencies = dependencies,
-            outputs = outputs,
-        ),
-    )
+load(
+    ":target_properties.bzl",
+    "process_defines",
+    "process_dependencies",
+    "process_modulemaps",
+    "process_sdk_links",
+    "process_swiftmodules",
+)
+load(
+    ":top_level_targets.bzl",
+    "process_top_level_properties",
+    "process_top_level_target",
+    "should_bundle_resources",
+    "should_include_outputs",
```

Comment:
These two should also be in `target_properties.bzl`, as they are used by library targets as well.